client/containers/Pub/PubDocument/PubFileImport/FileImportDialog.tsx

```diff
@@ -186,8 +186,15 @@ const FileImportDialog = ({
 			const missingCitations = warnings
 				.filter((w) => w.type === 'missingCitation')
 				.map((w) => w.id);
-
-			if (missingImages.length > 0 || missingCitations.length > 0) {
+			const pandocWarnings = pandocInfos.split('\n').map((info) => {
```

Comment:
React will complain that each of these children needs a `key` prop, which could just be the index from the second parameter of the map callback. Alternatively, though, you can accomplish the effect of splitting this at newlines by using the `white-space: pre` or `white-space: pre-wrap` (or maybe just `pre` given the width of the container) on an enclosing element.

Reply:
wow how did I not know pre was a thing?!